Movie exception and new /movie syntax

zedchance opened the pull request:
The `/movie` endpoint has a new simpler syntax. For example if searching for Titanic:

```
localhost:8080/movie/titanic
```

This also handles the 500 error encountered when the Movie isn't found.

zedchance committed:
Change /movie URI and handle MovieNotFoundException

The /movie endpoint now uses the syntax of /movie/title.
For example, /movie/titanic instead of previous /movie?title=titanic

When FetchFromCSV doesn't find a movie, a MovieNotFoundException is thrown.
This is handled by the Controller and an error message and 404 status code is returned.
The Error class is a simple object that holds the message and status code.

Closes #9

zedchance committed:
Update README with endpoint usage and examples

vgalbraith reviewed (approved):
Interesting to note that the formatting on spaces works differently now, used to be if we wanted "Knives Out" then putting in "Knives+Out" would work but now doesn't
Instead "Knives%20Out" does the trick. (This may have also worked before, not sure)

zedchance commented:
The %20 worked before with the ?title= syntax. Interesting that the + doesn't work with this...

This is something that whoever calls the API has to handle (URL encode).
@harnoorSingh99, when you hook up the UI make sure that you keep that in mind.

zedchance committed:
Override default /error endpoint

This now displays a "Endpoint not found" message when the endpoint
doesn't exist or isn't a thrown exception to another endpoint.

zedchance commented:
That last commit handles endpoints that don't exist like: `/asdfasdf`
